skyportal/utils/onboarding.py

```diff
@@ -0,0 +1,71 @@
+import datetime
+from social_core.exceptions import AuthTokenError
+from ..models import DBSession, User, Group, GroupUser, Invitation
+from baselayer.app.env import load_env
+
+
+env, cfg = load_env()
+
+USER_FIELDS = ['username', 'email']
+
+
+def create_user(strategy, details, backend, user=None, *args, **kwargs):
+    invite_token = strategy.session_get("invite_token")
+
+    if invite_token is not None:
+        try:
+            n_days = int(cfg["invitations.days_til_expiry"])
+        except ValueError:
+            raise ValueError(
+                "Invalid (non-integer) value provided for "
+                "invitations.days_til_expiry in config file."
+            )
+        invitation = Invitation.query.filter(Invitation.token == invite_token).first()
+        if invitation is None:
+            return
+
+        cutoff_datetime = datetime.datetime.now() - datetime.timedelta(days=n_days)
+        if invitation.created_at < cutoff_datetime:
+            raise AuthTokenError("Invite token has expired.")
+        user = User(
+            username=details["username"],
+            contact_email=details["email"],
+            first_name=details["first_name"],
+            last_name=details["last_name"],
+        )
+        DBSession().add(user)
```

Comment:
Should probably delete the invite token as soon as it's used to create a user; otherwise someone could pass their invite email on after they auth. This wont solve all edge cases but it's probably a good to protect against.

Reply:
@profjsb how about just marking it used/deactivated? We may still want a record of who invited whom / granted them access.